firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionDetailsViewController.swift

```diff
@@ -0,0 +1,280 @@
+
+        let viewCertificatesConstraints = [
+            viewCertificatesButton.leadingAnchor.constraint(
+                equalTo: view.leadingAnchor,
+                constant: TPMenuUX.UX.horizontalMargin
+            ),
+            viewCertificatesButton.trailingAnchor.constraint(
+                equalTo: view.trailingAnchor,
+                constant: -TPMenuUX.UX.horizontalMargin
+            ),
+            viewCertificatesButton.topAnchor.constraint(
+                equalTo: verifiedByView.bottomAnchor,
+                constant: TPMenuUX.UX.TrackingDetails.viewCertButtonTopDistance
+            ),
+        ]
+
+        constraints.append(contentsOf: viewCertificatesConstraints)
+    }
+
+    // MARK: Accessibility
+    private func setupAccessibilityIdentifiers() {
+        headerView.accessibilityIdentifier = model.headerViewA11yId
+        view.accessibilityIdentifier = model.detailsViewA11yId
+        baseView.accessibilityIdentifier = model.containerViewA11Id
+        connectionView.accessibilityIdentifier = model.connectionViewA11yId
+    }
+
+    // MARK: View Transitions
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        adjustLayout()
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        coordinator.animate(alongsideTransition: { _ in
+            self.adjustLayout()
+        }, completion: nil)
+    }
+
+    private func adjustLayout() {
+        let iconSize = TPMenuUX.UX.iconSize
+        connectionView.lockImageHeightConstraint?.constant = min(
+            UIFontMetrics.default.scaledValue(for: iconSize), 2 * iconSize
+        )
+
+        view.setNeedsLayout()
+        view.layoutIfNeeded()
+    }
+
+    private func updateViewDetails() {
+        headerView.siteTitleLabel.text = model.topLevelDomain
+        connectionView.connectionStatusLabel.text = model.connectionStatusMessage
+        let certificateVerifier =  String(format: .Menu.EnhancedTrackingProtection.connectionVerifiedByLabel,
+                                          model.topLevelDomain) // to be updated with the certificate verifier
+        verifiedByView.configure(verifiedBy: certificateVerifier)
+        viewCertificatesButton.setTitle(model.viewCertificatesButtonTitle, for: .normal)
+    }
+
+    // MARK: - Actions
+    @objc
+    func closeButtonTapped() {
+        self.dismiss(animated: true)
+    }
+
+    @objc
+    func viewCertificatesTapped() {
+    }
+
+    private func currentTheme() -> Theme {
+        return themeManager.getCurrentTheme(for: windowUUID)
+    }
+}
+
+// MARK: - Themable
+extension TrackingProtectionDetailsViewController {
+    func applyTheme() {
+        let theme = currentTheme()
+        overrideUserInterfaceStyle = theme.type.getInterfaceStyle()
```

Comment:
curious - why do we add this override?

Reply:
it's since the time we weren't sure if we'll use the native navigation bar or create our custom navBar, since this affects the colors for the native bar on theme changes.